jgo/jgo.py

```diff
@@ -717,15 +719,18 @@ def run(parser, argv=sys.argv[1:], stdout=None, stderr=None):
     repositories = config["repositories"]
     shortcuts = config["shortcuts"]
 
-    endpoint_index = find_endpoint(argv, shortcuts)
-    if endpoint_index == -1:
-        raise HelpRequested(
-            argv
-        ) if "-h" in argv or "--help" in argv else NoEndpointProvided(argv)
+    if "-h" in argv or "--help" in argv:
+        raise HelpRequested(argv)
+
+    args, unknown = parser.parse_known_args(argv)
+
+    if not args.endpoint:
+        raise NoEndpointProvided(argv)
+    if args.endpoint in shortcuts and not Endpoint.is_endpoint(args.endpoint):
+        raise NoEndpointProvided(argv)
 
-    args, unknown = parser.parse_known_args(argv[:endpoint_index])
```

Comment:
I've found a regression caused by this.  If the program args contains hyphenated args, e.g. `["com.pinterest:ktlint", "-F", "/c/path/to/file.kt"]` , the hyphenated arg becomes a JVM arg.

Reply:
A workaround for this is for people to use `--` , i.e. `["com.pinterest:ktlint", "--", "-F", "/c/path/to/file.kt"]`
Requiring that workaround would mean this is a breaking change.
Looking for other solutions ...

Reply:
Using `--` isnt enough.
`["--add-opens", "java.base/java.lang=ALL-UNNAMED", "com.pinterest:ktlint", "--", "-F", "/c/path/to/file.kt"]` ends up with jvm arg `["--add-opens"]` and endpoint `java.base/java.lang=ALL-UNNAMED`. :-(